main.tf

```diff
@@ -0,0 +1,116 @@
+terraform {
+  required_providers {
+    mongodbatlas = {
+      source = "mongodb/mongodbatlas"
+    }
+  }
+}
+
+resource "random_password" "password" {
+  length  = 16
+  special = false
+}
+
+locals {
+  database_users_default = tolist([{
+    username = "defaultUser"
+    password = tostring(random_password.password.result)
+    roles = tolist([{
+      database = "admin"
+      role     = "atlasAdmin"
+    }])
+    scopes = tolist([])
+  }])
+  database_users = length(var.database_users) == 0 ? local.database_users_default : var.database_users
+}
```

Comment:
If the dev provides a database user, we use the info provided. If not, we auto generate a password and give a default value for the username and the role.